### ISSUE TITLE
Add ArraySpeciesCreate abstract operation?

### ISSUE THREAD

**Issue by UltCombo** [domenic/especially#7]
Would it be desirable to add the [`ArraySpeciesCreate` abstract operation](http://www.ecma-international.org/ecma-262/6.0/#sec-arrayspeciescreate)?

I had started working on a PR to add this function, but then got stuck in the realms part. Looks like it is not possible to get the running execution context’s Realm nor a given object's [[Realm]] internal slot, correct?

Is there an approximation we can make for this operation?

**Comment by domenic** [comment]
Sorry for the delay.

I think it's probably best to just skip that part (maybe leaving a to-do).

Also, I am not actively using especially any more, so I'll add you to the repo and you can work on it and address any other issues you encounter, if you'd like :)

**Comment by UltCombo** [comment]
> I think it's probably best to just skip that part (maybe leaving a to-do).

Sounds good to me. So, can we assume that everything belongs to the same domain for now?

> I'll add you to the repo and you can work on it and address any other issues you encounter, if you'd like :)

That would be great. :smile:
I'm slightly short on free time lately, but I should be able to help a bit.

**Comment by domenic** [comment]
> So, can we assume that everything belongs to the same domain for now?

Yeah, I think that's really the only way to do things, since there's no way to reveal a function's realm in the language. I mean, maybe you could try testing if it's instanceof a certain global's Function.prototype, but that's not really the same thing.
